### DISCUSSION TITLE
DrawImage: Cannot draw image because the source image does not overlap the target image

### DISCUSSION THREAD

**Discussion by Raildex** [SixLabors/ImageSharp#2593] [Q&A]
I have two images,
one of the images should be cropped to a little portion and this portion should be drawn onto another image.
```cs
var orig = SixLabors.ImageSharp.Image.LoadPixelData<Bgra32>(p.Texture.Image.ToByteArray(), p.Texture.Image.Width, p.Texture.Image.Height);
orig.Mutate(ctx => ctx.Crop(new SixLabors.ImageSharp.Rectangle(x, y, width, height)));
var target = images[p.Page];
target.Mutate(ctx => ctx.DrawImage(orig, new SixLabors.ImageSharp.Rectangle(destX, destY, width, height), 1));
```

The last line throws an Exception in `DrawImage`
> DrawImage: Cannot draw image because the source image does not overlap the target image

I don't really now why though.

The debugger says the following:
<HTML>
<head>
</head>
<body>

  | Name | Value | Type
-- | -- | -- | --
▶ | target | {Image<RgbaVector>: 1024x1024} | SixLabors.ImageSharp.Image<SixLabors.ImageSharp.PixelFormats.RgbaVector>
▶ | orig | {Image<Bgra32>: 64x64} | SixLabors.ImageSharp.Image<SixLabors.ImageSharp.PixelFormats.Bgra32>
  | x | 0 | int
  | y | 1984 | int
  | width | 64 | int
  | height | 64 | int
  | destX | 88 | int
  | destY | 0 | int

</body>
</HTML>

As you can see, `target` is bigger than `destX + width` and `destY + height`
What is the issue here?

**Comment by JimBobSquarePants** [comment]
What version are you using?

**Reply by Raildex** [reply]
I am using version 2.0.0

**Reply by JimBobSquarePants** [reply]
See #2474 the fix will be released for v3.1
